src/main.rs

```diff
@@ -573,6 +722,131 @@ fn on_snapshot_button_clicked(
+    if record_button.get_active() {
+        // If we already have a record-bin (i.e. we still finish the previous one)
+        // just return for now and deactivate the button again
+        if pipeline.get_by_name("record-bin").is_some() {
+            record_button.set_state_flags(
+                record_button.get_state_flags() & !gtk::StateFlags::CHECKED,
+                true,
+            );
+            return;
+        }
+
+        let (bin_description, extension) = match settings.record_format {
+            RecordFormat::H264Mp4 => ("name=record-bin queue ! videoconvert ! x264enc ! video/x-h264,profile=baseline ! mp4mux ! filesink name=sink", "mp4"),
+            RecordFormat::Vp8WebM => ("name=record-bin queue ! videoconvert ! vp8enc ! webmmux ! filesink name=sink", "webm"),
+        };
+
+        let bin = match gst::parse_bin_from_description(bin_description, true) {
+            Err(err) => {
+                // TODO error dialogs
+                eprintln!("Failed to create recording pipeline: {}", err);
+                return;
+            },
+            Ok(bin) => bin,
+        };
+
+        // Get our file sink element by its name and set the location where to write the recording
+        let sink = bin.get_by_name("sink").unwrap();
+        let mut filename = settings.record_directory.clone();
+        let now = Local::now();
+        filename.push(format!(
+            "{}.{}",
+            now.format("Recording %Y-%m-%d %H:%M:%S"),
+            extension
+        ));
+        // All strings in GStreamer are UTF8, we need to convert the path to UTF8
+        // which in theory can fail
+        sink.set_property("location", &(filename.to_str().unwrap()))
+            .unwrap();
+
+        // First try setting the recording bin to playing: if this fails
+        // we know this before it potentially interferred with the other
+        // part of the pipeline
+        if let Err(_) = bin.set_state(gst::State::Playing).into_result() {
+            // TODO error dialogs
+            eprintln!("Failed to start recording bin");
+            return;
+        }
+
+        // Add the bin to the pipeline. This would only fail if there was already
+        // a bin with the same name, which we ensured can't happen
+        pipeline.add(&bin).unwrap();
+
+        // Get our tee element by name, request a new source pad from it and
+        // then link that to our recording bin to actually start receiving data
+        let tee = pipeline.get_by_name("tee").unwrap();
+        let srcpad = tee.get_request_pad("src_%u").unwrap();
+        let sinkpad = bin.get_static_pad("sink").unwrap();
+
+        // If linking fails, we just undo what we did above
+        if let Err(err) = srcpad.link(&sinkpad).into_result() {
+            // TODO error dialogs
+            eprintln!("Failed to link recording bin: {}", err);
+            pipeline.remove(&bin).unwrap();
+            let _ = bin.set_state(gst::State::Null);
```

Comment:
Is it returning an error or something?

Reply:
Yeah, but here we're in an error path anyway so any further error should be just ignored. Will add a comment, thanks!